src/common/object-view.ts

```diff
@@ -0,0 +1,35 @@
+import { stripIndent } from 'common-tags';
+import { DateTime } from 'luxon';
+import { RequireExactlyOne } from 'type-fest';
+import { ID } from './id-field';
+
+interface ObjectViewTypeMap {
+  active: true;
+  deleted: true;
+  changeset: ID | undefined;
+  beforeChangeset: ID | undefined;
```

Comment:
I think these should be required
```suggestion
  changeset: ID;
  beforeChangeset: ID;
```